aten/src/ATen/native/sparse/SparseCsrTensorMath.cpp

```diff
@@ -1275,10 +1334,11 @@ Tensor _sparse_csr_sum_cpu(const Tensor& input, IntArrayRef dims_to_sum, bool ke
   Tensor input_ = input.to(dtype_);
```

Comment:
I suppose that here `to` is for user specified `dtype`, should keep here.

Reply:
Type conversion to `dtype` is applied to input values only, however, there is a copy of indices as a side-effect from the `to` call which is undesired.

So, here we need to create a temporary input tensor using the indices of the input without the copy. So, instead of `Tensor input_ = input.to(dtype_);` have
```c++
Tensor compressed_indices, plain_indices;
std::tie(compressed_indices, plain_indices) = at::sparse_csr::getCompressedPlainIndices(input);
Tensor input_ = at::native::_sparse_compressed_tensor_unsafe(
      compressed_indices,
      plain_indices,
      input.values().to(dtype_),
      input.sizes(),
      dtype_,
      input.layout(),
      input.device());
```

In fact, this is needed elsewhere as well and it would be nice to have the corresponding type conversion tool available in aten/src/ATen/SparseCsrTensorUtils.h

Reply:
Done